src/plugins/resolver/mutex.c

```diff
@@ -3,5 +3,8 @@
 #include <pthread.h>
 
 // every resolver should use the same mutex
-pthread_mutex_t elektra_resolver_mutex =
-	PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
```

Comment:
Actually I think this won't work. The mutex won't be recursive, that means multiple nested KDB instances won't work anymore (but will deadlock).

If there is no equivalent for PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP on apple, we need an ifdef so that the problem only exists on apple.